File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int n = s.size();
        vector<int> sFreq(26, 0);
        for(int i=0; i<n; i++){
            sFreq[s[i] - 'a']++;
        }
        for(int i=0; i<n; i++){
            if(sFreq[s[i] - 'a'] == 1){
                return i;
            }
        }
        return -1;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int n = s.length();
        int sFreq[] = new int[26];
        for(int i=0; i<n; i++){
            sFreq[s.charAt(i) - 'a']++;
        }
        for(int i=0; i<n; i++){
            if(sFreq[s.charAt(i) - 'a'] == 1){
                return i;
            }
        }
        return -1;
    }
}